src/main/java/com/minecrafttas/common/server/Server.java

```diff
@@ -45,19 +45,10 @@ public Server(int port, PacketID[] packetIDs) throws Exception {
 
 			@Override
 			public void completed(AsynchronousSocketChannel clientSocket, Object attachment) {
-				cleanClients();
 				clients.add(new Client(clientSocket, packetIDs));
 				serverSocket.accept(null, this);
 			}
 
-			private void cleanClients() {
-				for(Client client : clients) {
-					if(client.isClosed()) {
-						clients.remove(client);
-					}
-				}
-			}
-
```

Comment:
for the future,
`clients.removeIf(Socket::isClosed);`
is a pretty neat way of doing this